core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -653,40 +652,33 @@ public void closeUfsBlock(long sessionId, long blockId)
   }
 
   @Override
-  public BlockReader createBlockReader(BlockReadRequest request) throws
-      BlockDoesNotExistException, IOException {
-    long sessionId = request.getSessionId();
-    long blockId = request.getId();
+  public BlockReader createBlockReader(long sessionId, long blockId, long offset,
+      boolean positionShort, Protocol.OpenUfsBlockOptions options) throws IOException
+  {
     RetryPolicy retryPolicy = new TimeoutRetry(UFS_BLOCK_OPEN_TIMEOUT_MS, Constants.SECOND_MS);
-    while (retryPolicy.attempt()) {
-      try {
-        BlockReader reader = createLocalBlockReader(sessionId, blockId, request.getStart());
-        Metrics.WORKER_ACTIVE_CLIENTS.inc();
-        return reader;
-      } catch (BlockDoesNotExistException e) {
-        boolean checkUfs =
-            request.isPersisted() || (request.getOpenUfsBlockOptions() != null && request
-                .getOpenUfsBlockOptions().hasBlockInUfsTier() && request.getOpenUfsBlockOptions()
-                .getBlockInUfsTier());
-        if (!checkUfs) {
-          throw e;
+    if (hasBlockMeta(blockId)) {
+      while (retryPolicy.attempt()) {
```

Comment:
I think the timeout retry is broken as before since we never retry(success or throw exception). But we are not sure whether we want to retry since all exception are wrapped in IOException...

Reply:
Good catch, I think my previous refactor misplaced the while loop.🤦‍♀️

Reply:
Hmm, I think even when the while loop is wrapping the whole function what you said is still true. We didn't swallow any exception so the retry will never be effective. IntelliJ actually is complaining about this as well.